A problem about "show more" button (fixes #1630)

Ashnidh opened the pull request:
I have fixed issue #1630 with this PR. 
Kindly check the correct translation of "See less" in Japanese as I used google translator for it.
I have fixed it for Spanish too.
I have added "Show more" translation for Hindi which also works fine after the fixes.

walterbender commented:
I don't understand what you are doing here.
(1) any changes to localization.inii should be made via the po files, e.g., ja.po
(2) can you explain the intention of the changes to helper.js?

Also, if occurs to me that the English is wrong: it should be "View fewer" or "Show fewer". It might be even more clear if we say "View fewer tags" (and "Show more tags").

Ashnidh commented:
When language is set as English, "View More" toggles to "View Less" on click. 

![image](https://user-images.githubusercontent.com/44440524/49956103-6d2a5400-ff2a-11e8-83e5-af89ed29becd.png)

The issue #1630 states that in Japanese version it does not happen. (even not in any other language version except English)

I have tried to fix it. To test it I have added the Hindi translation of "View more" in localization.ini (Now I have changed it in hi.po)

When I added the following strings in helper.js, it works fine in Hindi, Spanish and Japanese versions like english one.
		toggleText('view-more-chips','और दिखाएं','कम दिखाएं');
		toggleText('view-more-chips','Ver más','Ver menos');
		toggleText('view-more-chips','もっと見る','あまり見ない');

I have changed the English "VIEW LESS" to "VIEW FEWER" as per your suggestions.

aust-n commented:
@walterbender left some comments in #1649 (by accident) which, I think, were meant for this issue :

The View More and View Less strings are not marked for translation, which is probably the root cause of the problem.

Should be

`toggleText('view-more-chips',_('View More'), _('View Fewer'));`

aust-n commented:
Is it mandatory for strings enclosed to have a "po" counterpart in order to be translated - i.e. must you specify every instance of a __(string)_ in the po files?

I ask because I've been poking around [here](https://github.com/sugarlabs/musicblocks/compare/master...aust-n:show-more-planet?expand=1) and it seems either the po files need to be generated or there is a bug somewhere else.

walterbender commented:
The flow is as follows:

Any string enclosed in _('YOUR STRING') in Javascript will end up in the POT file.
We generate (and update) PO files from the POT file.
Translators translate strings in the PO files.
A separate process is used generate the localization.ini file from the PO files.

aust-n commented on planet/js/helper.js:
```diff
@@ -138,8 +138,8 @@ $(document).ready(function() {
+        toggleText('view-more-chips','View More','View Fewer');
+		toggleText('view-more-chips','और दिखाएं','कम दिखाएं');
 		toggleText('view-more-chips','Ver más','Ver menos');
 		toggleText('view-more-chips','もっと見る','あまり見ない');
```
Translations in Music Blocks are handled centrally via the generation of PO files (instead of manually translating phrases in the 'main' code as you have done, you should make the string translatable via the surrounding the string with _( string ) as Walter mentioned :P). 
As I understand, this 'PO' method is pretty neat, in that it's somewhat simple and yet is very scalable; it also seems to make adding new languages, and their translations, straightforward.

Ashnidh commented:
Thank you @walterbender.
Thank you @aust-n for fixing it in #1652 (so I am closing the PR)
